pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -24,15 +24,19 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/statushandler"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/watch"
 	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 // AtlasProjectReconciler reconciles a AtlasProject object
 type AtlasProjectReconciler struct {
 	client.Client
+	watch.ResourceWatcher
```

Comment:
Is there a reason this can't be demoted to a field? I find regular fields to be way safer and more obvious than embedding in most cases.

Reply:
I agree with you in general - especially for the `client.Client`!
Here though I intended to conform to the "is-a" model more - to point out that the reconciler "is a" resource watcher itself and pretends to contain that watched map internally...

WDYT?

Reply:
I agree with you in general - especially for the `client.Client`!
Here though I intended to conform to the "is-a" model more - to point out that the reconciler "is a" resource watcher itself and pretends to contain that watched map internally...

WDYT?

Reply:
Not something I'm willing to die for - go ahead :smiley: